### DISCUSSION TITLE
Best way to strip empty options from string dropdown

### DISCUSSION THREAD

**Discussion by genox** [sanity-io/sanity#3986] [Sanity Studio Support]
I use a lot of dropdown string fields and usually define a default selected option for editors.

Is it possible to disable the undefined option without having to hack a custom component or other trickery? It's confusing and unintuitive.

<img width="344" alt="Screenshot 2022-12-16 at 09 16 58" src="https://user-images.githubusercontent.com/5611407/208057370-d6b6ac40-e83f-4226-8023-27c662440cdf.png">

I couldn't find a configuration option to make that happen. Or could we add this as a feature?

**Comment by plsrd** [comment]
The empty option exists to allow the field to be unset. If you have a specific case where you always need that field to have a value, you can either use validation or a custom input component.

**Comment by genox** [comment]
Thanks for your answer.

I get that. But look at it from this perspective: If a component is defined to be always required using `Validation.required()` _and_ has an `initialValue` set, unsetting is specifically not wanted, so it would improve UX to simply not display an option that only has the potential to cause an error. That's what I mean with "unintuitive". A flag would be sufficient to just hide an empty option if so desired.
